src/frontend/admin/src/components/presentational/discount/DiscountSelect.tsx

```diff
@@ -0,0 +1,262 @@
+    id: "components.presentational.discount.DiscountSelect.discountAmountHelperText",
+    defaultMessage: "Enter a fixed amount in euros",
+    description: "The helper text for the amount field",
+  },
+  discountExclusiveHelperText: {
+    id: "components.presentational.discount.DiscountSelect.discountExclusiveHelperText",
+    defaultMessage: "You can only set either a rate or an amount, not both",
+    description:
+      "The helper text explaining that rate and amount are exclusive",
+  },
+});
+
+type Props = {
+  value?: string | null;
+  onChange: (discountId: string | null) => void;
+  error?: boolean;
+  helperText?: string;
+};
+
+export function DiscountSelect({ value, onChange, error, helperText }: Props) {
+  const intl = useIntl();
+  const [discounts, setDiscounts] = useState<Discount[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [newDiscount, setNewDiscount] = useState<DTODiscount>({
+    amount: null,
+    rate: null,
+  });
+  const [formError, setFormError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchDiscounts = async () => {
+      try {
+        const response = await DiscountRepository.getAll();
+        setDiscounts(response.results);
+      } catch (fetchError) {
+        setFormError(`Failed to fetch discounts: ${fetchError}`);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchDiscounts();
+  }, []);
+
+  const handleOpenDialog = () => {
+    setDialogOpen(true);
+  };
+
+  const handleCloseDialog = () => {
+    setDialogOpen(false);
+    setNewDiscount({ amount: null, rate: null });
+    setFormError(null);
+  };
+
+  const handleCreateDiscount = async () => {
+    // Validate that either rate or amount is set, but not both
+    if (
+      (newDiscount.rate === null && newDiscount.amount === null) ||
+      (newDiscount.rate !== null && newDiscount.amount !== null)
+    ) {
+      setFormError(intl.formatMessage(messages.discountExclusiveHelperText));
+      return;
+    }
+
+    try {
+      if (newDiscount.rate !== null) {
+        newDiscount.rate /= 100; // Convert to decimal
+      }
+      const createdDiscount = await DiscountRepository.create(newDiscount);
+      setDiscounts([...discounts, createdDiscount]);
+      onChange(createdDiscount.id);
+      handleCloseDialog();
+    } catch (fetchError) {
+      setFormError(`Failed to create discount: ${fetchError}`);
+    }
+  };
+
+  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue =
+      e.target.value === "" ? null : parseInt(e.target.value, 10);
+    setNewDiscount({ ...newDiscount, amount: inputValue, rate: null });
+  };
+
+  const handleRateChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue =
+      e.target.value === "" ? null : parseFloat(e.target.value);
+    setNewDiscount({ ...newDiscount, rate: inputValue, amount: null });
+  };
```

Comment:
To manage FormState we are using React Hook Form which helps to manage form validation through a validation schema declared through `yup`. If I remember well there are already some form component ready to be integrated with React Hook Form. Those kind of components are prefixed by `RHF`